src/prefect/task_engine.py

```diff
@@ -394,6 +444,16 @@ def handle_exception(self, exc: Exception) -> None:
             )
             self.set_state(state)
 
+    async def handle_exception_async(self, exc: Exception) -> None:
```

Comment:
yea same as above, once we do one async method it begins forcing duplication in other places which is what we want to avoid - we could consider `sync_compatible` on the async versions of these and we call them with `sync=True` in the sync engine.